src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinSourceEdgeWriter.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.spring.data.gremlin.conversion;
+
+import com.microsoft.spring.data.gremlin.annotation.EdgeFrom;
+import com.microsoft.spring.data.gremlin.annotation.EdgeTo;
+import com.microsoft.spring.data.gremlin.common.Constants;
+import com.microsoft.spring.data.gremlin.mapping.GremlinPersistentEntity;
+import org.springframework.data.mapping.PersistentProperty;
+import org.springframework.data.mapping.model.ConvertingPropertyAccessor;
+import org.springframework.lang.NonNull;
+import org.springframework.util.Assert;
+
+import java.lang.reflect.Field;
+
+public class GremlinSourceEdgeWriter extends BasicGremlinSourceWriter implements GremlinSourceWriter {
+
+    public GremlinSourceEdgeWriter(@NonNull Object domain) {
+        super(domain);
+    }
+
+    private String getPersistentEntityId(Object domain) {
+        Assert.notNull(domain, "domain class should not be null");
+
+        return new BasicGremlinSourceWriter(domain).getPersistentEntityId();
```

Comment:
Why instantiating a parent class here?

Reply:
Actually, it requires the id of another domain. This can be replaced by GremlinEntityInformation(domain).getIdField().toString() here. That makes more clear.